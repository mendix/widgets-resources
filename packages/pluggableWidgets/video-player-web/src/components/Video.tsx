import { Component, createElement } from "react";

import { Dailymotion } from "./Dailymotion";
import { Html5 } from "./Html5";
import { Vimeo } from "./Vimeo";
import { Youtube } from "./Youtube";

export interface VideoPlayerProps {
    url?: string;
    style?: object;
    poster?: string;

    autoStart: boolean;
    showControls: boolean;
    loop: boolean;
    muted: boolean;
    aspectRatio: boolean;

    preview?: boolean;
}

export class Video extends Component<VideoPlayerProps> {
    private readonly handleHtml5PlayerRender = this.renderHtml5Player.bind(this);
    private readonly handleYoutubePlayerRender = this.renderYoutubePlayer.bind(this);
    private readonly handleVimeoPlayerRender = this.renderVimeoPlayer.bind(this);
    private readonly handleDailymotionPlayerRender = this.renderDailymotionPlayer.bind(this);

    render(): JSX.Element {
        const url = this.props.url || "";
        if (Youtube.canPlay(url)) {
            return this.handleYoutubePlayerRender(url);
        } else if (Vimeo.canPlay(url)) {
            return this.handleVimeoPlayerRender(url);
        } else if (Dailymotion.canPlay(url)) {
            return this.handleDailymotionPlayerRender(url);
        }
        return this.handleHtml5PlayerRender(url);
    }

    private renderHtml5Player(url: string): JSX.Element {
        return (
            <Html5
                showControls={this.props.showControls}
                autoPlay={this.props.autoStart}
                muted={this.props.muted}
                loop={this.props.loop}
                poster={this.props.poster}
                url={url}
                aspectRatio={this.props.aspectRatio}
            />
        );
    }

    private renderYoutubePlayer(url: string): JSX.Element {
        return (
            <Youtube
                url={url}
                showControls={this.props.showControls}
                autoPlay={this.props.autoStart}
                muted={this.props.muted}
                loop={this.props.loop}
                aspectRatio={this.props.aspectRatio}
            />
        );
    }

    private renderVimeoPlayer(url: string): JSX.Element {
        return (
            <Vimeo
                url={url}
                autoPlay={this.props.autoStart}
                muted={this.props.muted}
                loop={this.props.loop}
                aspectRatio={this.props.aspectRatio}
            />
        );
    }

    private renderDailymotionPlayer(url: string): JSX.Element {
        return (
            <Dailymotion
                url={url}
                autoPlay={this.props.autoStart}
                muted={this.props.muted}
                showControls={this.props.showControls}
                aspectRatio={this.props.aspectRatio}
            />
        );
    }
}
