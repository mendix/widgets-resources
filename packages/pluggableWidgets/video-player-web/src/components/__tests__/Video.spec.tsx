import { createElement } from "react";

import { Video } from "../Video";
import { create } from "react-test-renderer";

describe("Video Player", () => {
    it("Renders the structure of youtube tags", () => {
        const player = create(
            <Video
                url="http://youtube.com/video/123456"
                poster=""
                autoStart={false}
                showControls={false}
                loop={false}
                muted={false}
                aspectRatio={false}
            />
        ).toJSON();

        expect(player).toMatchSnapshot();
    });

    it("Renders the structure of vimeo tags", () => {
        const player = create(
            <Video
                url="http://vimeo.com/123456"
                poster=""
                autoStart={false}
                showControls={false}
                loop={false}
                muted={false}
                aspectRatio={false}
            />
        ).toJSON();

        expect(player).toMatchSnapshot();
    });

    it("Renders the structure of dailymotion tags", () => {
        const player = create(
            <Video
                url="http://dailymotion.com/123456"
                poster=""
                autoStart={false}
                showControls={false}
                loop={false}
                muted={false}
                aspectRatio={false}
            />
        ).toJSON();

        expect(player).toMatchSnapshot();
    });

    it("Renders the structure of html5 player tags", () => {
        const player = create(
            <Video
                url="http://ext.com/video.mp4"
                poster=""
                autoStart={false}
                showControls={false}
                loop={false}
                muted={false}
                aspectRatio={false}
            />
        ).toJSON();

        expect(player).toMatchSnapshot();
    });
});
