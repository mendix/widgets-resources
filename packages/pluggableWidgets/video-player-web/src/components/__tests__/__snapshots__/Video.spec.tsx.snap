// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Player Renders the structure of dailymotion tags 1`] = `
<iframe
  allow="autoplay; fullscreen"
  allowFullScreen={true}
  className="widget-video-player-iframe"
  frameBorder="0"
  src="https://www.dailymotion.com/embed/video/123456?sharing-enable=false&autoplay=false&mute=false&controls=false"
>
  <div />
</iframe>
`;

exports[`Video Player Renders the structure of html5 player tags 1`] = `
<div
  className="widget-video-player-html5-container"
>
  <div
    className="video-error-label-html5"
  >
    The video failed to load :(
  </div>
  <video
    autoPlay={false}
    className="widget-video-player-html5"
    controls={false}
    height="100%"
    loop={false}
    muted={false}
    poster=""
    preload="auto"
  >
    <source
      onError={[Function]}
      onLoad={[Function]}
      src="http://ext.com/video.mp4"
      type="video/mp4"
    />
    <div />
  </video>
</div>
`;

exports[`Video Player Renders the structure of vimeo tags 1`] = `
<iframe
  allow="autoplay; fullscreen"
  allowFullScreen={true}
  className="widget-video-player-iframe"
  frameBorder="0"
  src="https://player.vimeo.com/video/123456?dnt=1&autoplay=0&muted=0&loop=0"
>
  <div />
</iframe>
`;

exports[`Video Player Renders the structure of youtube tags 1`] = `
<iframe
  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
  allowFullScreen={true}
  className="widget-video-player-iframe"
  frameBorder="0"
  src="http://youtube.com/video/123456"
>
  <div />
</iframe>
`;
